src/validators/auth.ts

```diff
@@ -112,3 +113,46 @@ export async function validateRefreshToken(
     }
   }
 }
+
+/**
+ * check for existing user
+ *
+ * @param  {Request}   req
+ * @param  {Response}   res
+ * @param  {NextFunction} next
+ * @returns {Promise}
+ */
+export async function checkUser(
+  req: Request,
+  res: Response,
+  next: NextFunction
+) {
+  try {
+    const mobile = req.body.mobile;
+    const email = req.body.email;
+
+    const [userMobile] = await userService.search({ mobile });
+    const [userEmail] = await userService.search({ email });
```

Comment:
Since awaiting for two responses will take time. The userEmail is searched only when the upper await is finished. 
So to overcome this and make a litte faster response time. We can use Promise.all and await that.
```suggestion

    const userMobilePromise = userService.search({ mobile });
    const userEmailPromise = userService.search({ email });
    
    const  [ [userMobile], [userEmail] ] = await Promise.all( [ userMobilePromise, userEmailPromise ] );
    
```

Reply:
done